Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -97,6 +97,187 @@ function Setup-Prerequisites
     Write-Host "All required modules are present." -ForegroundColor $([Constants]::MessageType.Update)
 }
 
+function Fetch-API {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string]$Method,
+        
+        [Parameter(Mandatory=$true)]
+        [string]$Uri,
+        
+        [Parameter(Mandatory=$false)]
+        [hashtable]$Body = @{},
+        
+        [Parameter(Mandatory=$false)]
+        [hashtable]$Headers = @{}
+    )
+
+    $cloudEnvironmentResourceManagerUrl = (Get-AzContext).Environment.ResourceManagerUrl
+    $accessToken = Get-AzAccessToken -ResourceUrl $cloudEnvironmentResourceManagerUrl
+    $authHeader = "Bearer " + $accessToken.Token
+    $Headers["Authorization"] = $authHeader
+    $Headers["Content-Type"] = "application/json"
+
+    try {
+        switch ($Method.ToUpper()) {
+            "GET" {
+                $response = Invoke-WebRequest -Uri $Uri -Method Get -Headers $Headers -UseBasicParsing -ErrorAction Stop
+            }
+            "POST" {
+                $jsonBody = $Body | ConvertTo-Json
+                $response = Invoke-WebRequest -Uri $Uri -Method Post -Headers $Headers -Body $jsonBody -UseBasicParsing -ErrorAction Stop
+            }
+            "PUT" {
+                $jsonBody = $Body | ConvertTo-Json
+                $response = Invoke-WebRequest -Uri $Uri -Method Put -Headers $Headers -Body $jsonBody -UseBasicParsing -ErrorAction Stop
+            }
+            "DELETE" {
+                $response = Invoke-WebRequest -Uri $Uri -Method Delete -Headers $Headers -UseBasicParsing -ErrorAction Stop
+            }
+            default {
+                throw "Unsupported HTTP method: $Method"
+            }
+        }
+
+        if ($response.StatusCode -eq 200 -or $response.StatusCode -eq 201) {
+            return $response.Content | ConvertFrom-Json
+        }
+        else {
+            throw "API call failed with status code $($response.StatusCode)"
+        }
+    }
+    catch {
+        Write-Error "Error occurred: $_"
+    }
+}
+
+function Remediate-VirtualMachines {
+    param (
+        [string]$subscriptionId,          # Subscription ID
+        [string]$reqMDCTier,              # Required MDC Tier (e.g., Standard, Free)
+        [string]$currentVulnerabilitySetting, # Current server vulnerability assessment setting (e.g., MdeTvm)
+        [bool]$isWDATPEnabled             # Whether WDATP is already enabled
+    )
+    try {
+        $virtualMachinePricing = Set-AzSecurityPricing -Name "VirtualMachines" -PricingTier $reqMDCTier
+    }
+    catch {
+        Write-Host "Failed to set pricing tier for VirtualMachines"
+        return $_.Exception.Message
+    }
+
+    if ($currentVulnerabilitySetting -ne "MdeTvm") {
+        # Inform user and get confirmation
+        $confirmation = Read-Host "The current vulnerability setting is $currentVulnerabilitySetting. Once changed to MdeTvm, you cannot roll back to the previous setting. Do you still want to remediate? (Y/N)"
```

Comment:
Hope, this rollback feasibility is verified properly.

Reply:
Yes Rajat, tested several ways to roll back using API as well.
[BRS analysis](https://microsoft.sharepoint.com/teams/ISRMSAFETIndia/_layouts/OneNote.aspx?id=%2Fteams%2FISRMSAFETIndia%2FShared%20Documents%2FAzSK-AzureTenantSecurity&wd=target%28TBv14%20Controls.one%7CDA549D24-BF41-4A2E-9524-3538666076A6%2FBRS%20Azure_Subscription_Config_Enable_MicrosoftDefender_Servers_Trial%7C021C254B-C643-4D60-9F40-0C6F5BDA2498%2F%29
onenote:https://microsoft.sharepoint.com/teams/ISRMSAFETIndia/Shared%20Documents/AzSK-AzureTenantSecurity/TBv14%20Controls.one#BRS%20Azure_Subscription_Config_Enable_MicrosoftDefender_Servers_Trial&section-id={DA549D24-BF41-4A2E-9524-3538666076A6}&page-id={021C254B-C643-4D60-9F40-0C6F5BDA2498}&end)